### PR TITLE
health: remove all vimscript from nvim/health

### PR DESCRIPTION
**Pull request by tjdevries**
I'll have to do a bit more testing, but this is most of the work

### CODE REVIEW

File: runtime/lua/nvim/health.lua [hunk 1]
```diff
@@ -1,6 +1,23 @@
 local M = {}
 local health = require('vim.health')
 
+local lua_has_bool = function(key)
```
Comment (justinmk) [inline runtime/lua/nvim/health.lua:4]:
```suggestion
local fn_bool = function(key)
```

File: runtime/lua/nvim/health.lua [hunk 2]
```diff
@@ -139,240 +147,245 @@ local function check_config()
...
+    health.report_info(
+      vim.fn.printf(
+        'key_dc (kdch1) terminfo entry: `%s`',
+        (empty(kbs_entry) and '? (not found)' or kdch1_entry)
+      )
+    )
+  end
+
+  for env_var in ipairs({ 'XTERM_VERSION', 'VTE_VERSION', 'TERM_PROGRAM', 'COLORTERM', 'SSH_TTY' }) do
+    if exists('$' .. env_var) then
```
Comment (justinmk) [inline runtime/lua/nvim/health.lua:385]: vim.env ?

```suggestion
    if vim.env[env_var] then
```

**Comment by tjdevries** [comment]
resolved

**Comment by tjdevries** [comment]
@justinmk do i need to tag all of these functions with `@private` or something to make docs not fail? I'm not sure what happened there.

**Comment by tjdevries** [comment]
oh nvm, it was a flaky test

**Comment by tjdevries** [comment]
As far as I can tell, this should be OK

**Comment by przepompownia** [comment]
https://github.com/neovim/neovim/blob/307efe4906de9f0b7d8611ea36bddb85493c1447/runtime/lua/nvim/health.lua#L57-L60

If `init.lua` exists instead of `init.vim` missing user config file is still reported.

**Pull request merged**